browserfriend/llm/analyzer.py

```
@@ -0,0 +1,482 @@
+        browserfriend.llm.APIKeyError: If the API key is missing.
+    """
+    from browserfriend.llm import APIKeyError
+
+    config = get_config()
+    api_key = config.google_api_key
+    if not api_key or api_key == "your_google_api_key_here":
+        raise APIKeyError(
+            "Google API key not configured. "
+            "Set GOOGLE_API_KEY in your .env file. "
+            "Get a key at https://makersuite.google.com/app/apikey"
+        )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)